### DISCUSSION TITLE
monome-serialosc-client: NPE after s.connect("...")

### DISCUSSION THREAD

**Discussion by erhard3mem** [community/community#45054] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Bug

### Body

@jiparis

.... I know this is old, but.

`MonomeZeroconf s = new MonomeZeroconf();
  String[] monomes = s.getDevices(); 
  
  // connect
  MonomeDevice m = s.connect(monomes[0], "/myapp", 8000);
  // listen for grid events
  m.addListener(new GridListener() {
    @Override	
    public void press(int x, int y, int state) {
	  m.grid.set(x, y, state);				
    }
  });` 

java.lang.NullPointerException: Cannot invoke "MonomeDevice.addListener(Object)" because "m" is null

I guess this problem is coming my Ubuntu and JmDNS. Can you verify? 
I linked all the jar files to the execution environment.

**Comment by jge162** [comment]
@erhard3mem i suggest you do the following. Verify that the JmDNS library is included in the classpath and that it is compatible with your system.Ensure that your system supports multicast and that the multicast address is not blocked. Make sure that your Monome device is connected to the network and powered on. Verify that the Monome device is properly configured for service discovery.

**Reply by erhard3mem** [reply]
Thx for helping. 

in MonomeZeroconf.java: `ServiceInfo info = monomesMap.get(name); `

... info is null. 

Monome device is connected. 

`serialosc [m64-0383]: connected, server running on port 10379`

What do you mean exactly the device is properly configured for discovery?

I am also having some NodeJS applications which connect successfully the monome to do some stuff. (OT: But my problem was, the sound libraries in NodeJS are very slow, if you want to play an 1s-mp3 after a key press for example).

**Reply by jge162** [reply]
properly configured for service discovery, the device should have the correct settings configured in its firmware, such as the hostname, IP address, and port number. The settings can be configured using the serialosc tool.

**Reply by jge162** [reply]
@erhard3mem Regarding your NodeJS application, you can use the same Monome device in your application and other applications simultaneously as long as they don't use the same port.

NodeJS is a JavaScript runtime built on Chrome's V8 JavaScript engine and its libraries are widely used. There's a library called 'monome' which allows you to interact with Monome devices using NodeJS, you can check it out and see if it suits your needs better.

**Comment by queenofcorgis** [comment]
Thanks for posting in the GitHub Community, @erhard3mem ! 

We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category, the Issues category is solely related to conversations around the GitHub products Issues and Projects. I've gone ahead and moved it for you. Good luck!
